### PR TITLE
Enhancement of the carousel samples

### LINKED ISSUE #1577: Carousel samples for the new animations
### Description  
Think of a samples that will demonstrate the new perks of the carousel, like:
- new slide animations
- buttons navigation and indicator templates
- pan interactions
- everything else

### PR DESCRIPTION
**Pull request by HristoP96**
Closes #1577

### CODE REVIEW

File: src/app/layouts/carousel/carousel-no-navigation-sample/carousel-no-navigation-sample.component.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        animation: buynow 360ms cubic-bezier(0.445, 0.05, 0.55, 0.95) forwards;
+    }
+
+    .promo-price {
+        display: flex;
+        background: rgba(255,255,255,.55);
+        padding: 1rem;
+        border-bottom: transparent 1px solid;
+        justify-content: center;
+        filter: drop-shadow(1px 1px 1px rgba(0,0,0, .12));
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-no-navigation-sample/carousel-no-navigation-sample.component.scss]: Why don't you use box-shadow?

File: src/app/layouts/carousel/carousel-no-navigation-sample/carousel-no-navigation-sample.component.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    @keyframes buynow {
+        from {
+            opacity: .2;
+            transform: translateY(-25%);
+            filter: blur(4px);
+        }
+        to {
+            opacity: 1;
+            transform: translateY(0);
+            filter: blur(0);
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-no-navigation-sample/carousel-no-navigation-sample.component.scss]: This filter won't work in IE11.

File: src/app/layouts/carousel/carousel-base-sample/carousel-base-sample.component.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        height: 350px;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+    }
+
+    .indicator{
+        width: 12px;
+        height: 12px;
+        content: ' ';
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-base-sample/carousel-base-sample.component.scss]: Why do we use `content` here?

File: src/app/layouts/carousel/carousel-animations-sample/carousel-animations-sample.component.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    max-height: 600px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 100%;
+    padding: 10px;
+
+    img {
+        width: 100%;
+        object-fit: contain;
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-animations-sample/carousel-animations-sample.component.scss]: object-fit will not work in IE 11.

File: src/app/layouts/carousel/carousel-sample-1/carousel-sample-1.component.scss [hunk 5]
```diff
@@ -1,7 +1,16 @@
...
-}
-igx-carousel igx-icon.igx-icon{
-    color: #e41c77;
+
+:host ::ng-deep{
+
+    .igx-carousel__inner {
+        width: 100%;
+        text-align: center; 
+    }
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-sample-1/carousel-sample-1.component.scss]: Don't modify the inner part of the carousel directly. This is a big no no.

File: src/app/layouts/carousel/carousel-sample-1/carousel-sample-1.component.scss [hunk 6]
```diff
@@ -1,7 +1,16 @@
...
+
+    igx-carousel igx-icon.igx-icon{
```
Comment (simeonoff) [inline src/app/layouts/carousel/carousel-sample-1/carousel-sample-1.component.scss]: The `igx-icon.igx-icon` selector is a bit much. You can target the `igx-icon` only, as the igx-icon class is always set anyways.

**Comment by HristoP96** [comment]
@simeonoff requested changes are done

**Pull request merged**